safe-global/safe-react
# fix: throttle balance state syncer

## What it solves
Resolves #3176

## How this PR fixes it
- `blockPollingInterval` increased
- onboard's balance state syncer interval increased
- patched WC module removed

### `blockPollingInterval`
The `blockPollingInterval` has been increased from 4 seconds to an hour. This has no knock-on effect as we do not access the block details from the provider but from `Web3`. Confirmed by testing:

![image](https://user-images.githubusercontent.com/20442784/164224173-97d646c7-c761-4db0-b1bf-76afe2089ad1.png)

When connecting a Ledger or Trezor there will no longer be overpolling to the RPC. This has been tested on a Trezor T and Nano X, successfully creating a Safe on both.

Note: if the Trezor throws 'Forbidden key path', the ['Safety checks' need to be altered to 'Prompt' in the Trezor Suite](https://github.com/MetaMask/metamask-extension/issues/10390#issuecomment-974591474).

### Onboard's balance state syncer
Onboard's balance state syncer ([`walletInterface.balance.get`]([url](https://docs.blocknative.com/onboard-older-version#custom-wallet-modules))) is [called every 200ms](https://github.com/blocknative/web3-onboard/blob/v1-onboard/src/stores.ts#L278-L291).

We never access the balance of the connected wallet from onboard and has been patched, also to be an hour

### Patched WC module
As the balance state syncer has been patched, the custom WC module that was used to avoid excessive `eth_getBalance` requests has been removed in place of the official one instead.

## How to test it
- Connect a Ledger/Trezor directly to the Safe and observe no `eth_getBalance` over-polling.
- WalletConnect should also not over poll.
- Transactions should function as usual with all three.

## Screenshots
An example of network requests from a Trezor: (note that the initial wallet address selection modal will poll for the address of each available addresses on the device, but no more)

![trezor](https://user-images.githubusercontent.com/20442784/164276588-73039d5b-4d40-4bd1-ba08-69e3c1a61328.gif)
